src/Stratis.Bitcoin/Connection/PeerBanning.cs

```diff
@@ -112,23 +112,24 @@ public PeerBanning(IConnectionManager connectionManager, ILoggerFactory loggerFa
         public void BanPeer(IPEndPoint endpoint, int banTimeSeconds, string reason = null)
         {
             Guard.NotNull(endpoint, nameof(endpoint));
-
             this.logger.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(endpoint), endpoint, nameof(reason), reason);
 
+            reason = reason ?? "unknown";
+
             bool banPeer = true;
             NetworkPeer peer = this.connectionManager.ConnectedNodes.FindByEndpoint(endpoint);
             if (peer != null)
             {
                 ConnectionManagerBehavior peerBehavior = peer.Behavior<ConnectionManagerBehavior>();
                 if (!peerBehavior.Whitelisted)
                 {
-                    this.logger.LogDebug("Peer '{0}' banned for reason '{1}'.", endpoint, reason ?? "unknown");
+                    this.logger.LogDebug("Peer '{0}' banned for reason '{1}'.", endpoint, reason);
```

Comment:
This will not always be hit perhaps we should move this trace in to the if condition bellow?

Reply:
it looks fine here for me, where exactly do you mean to move it?

Reply:
If for what ever reason `NetworkPeer` is not in the collection we wont trace the ban.

I would put it here
```
 if (banPeer)
{		             
   this.logger.LogDebug("Peer '{0}' banned for reason '{1}'.", endpoint, reason);
   this.banStore.BanPeer(endpoint, this.dateTimeProvider.GetUtcNow().AddSeconds(banTimeSeconds));		                 this.banStore.BanPeer(endpoint, this.dateTimeProvider.GetUtcNow().AddSeconds(banTimeSeconds));
}		             
```